### PR TITLE
Remove the pyi_rth_win32comgenpy 

### PR DESCRIPTION
**Pull request by arossert**
Following this discussion https://github.com/orgs/pyinstaller/discussions/8309 it was decided to remove this hook

### CODE REVIEW

**Comment by bwoodsend** [comment]
Just needs a [news entry](https://pyinstaller.readthedocs.io/en/stable/development/changelog-entries.html). Not sure what it would be though. *Remove obsolete redirection of win32com's cache to a temporary directory*?

**Comment by rokm** [comment]
> Just needs a [news entry](https://pyinstaller.readthedocs.io/en/stable/development/changelog-entries.html). Not sure what it would be though. _Remove obsolete redirection of win32com's cache to a temporary directory_?

``` 
Remove run-time hook for ``win32com``, as per discussion in issue:`8309`.
``` 

should suffice, I think.

**Comment by arossert** [comment]
@rokm Should it be a feature, bugfix, or breaking?

**Comment by rokm** [comment]
> @rokm Should it be a feature, bugfix, or breaking?

I'd put it under `hooks`.

File: news/8313.hooks.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove run-time hook for ``win32com``, as per discussion in issue:`8309`.
```
Comment (bwoodsend) [inline news/8313.hooks.rst:1]: It's actually
```suggestion
Remove run-time hook for ``win32com``, as per discussion in :issue:`8309`.
```
but it's easier if you just leave it and I'll mop it up at release time.

**Review by bwoodsend — APPROVED** [review]

**Review by rokm — APPROVED** [review]

**Pull request merged**